src/de/mod.rs

```diff
@@ -2247,9 +2247,19 @@ where
 
                 // If we do not have skipped events, use effective reading that will
                 // not allocate memory for events
-                None => return self.reader.read_to_end(name),
+                None => {
+                    loop {
+                        self.reader.read_to_end(name)?;
```

Comment:
Could you add a comment about what the extra logic is for - with context it's not difficult to figure out, but it would help any uninitiated readers of this code.

Reply:
Added. Could you check that it is clearly enough? [Diff](https://github.com/tafia/quick-xml/compare/9e7dcaad8a19e89a648e8f35795fd49a388e4c15..ca7aa21ba111b28ca54ddc161c8302bba035904f)

Reply:
:+1: